pages/members/[id]/index.js

```diff
@@ -1,10 +1,45 @@
+import { getDataURL, getImgURL } from 'helper-functions/urls';
+import PropTypes from 'prop-types';
+import fetch from 'cross-fetch';
 import React from 'react';
-import { useRouter } from 'next/router';
 import Profile from 'components/member-profile';
+import NotFound from 'components/404page';
 
-const MemberProfile = () => {
-  let { query } = useRouter();
-  return <Profile id={query.id} />;
+const MemberProfile = ({ imageLink, data, errorMessage }) => {
+  if (errorMessage) {
+    return <NotFound />;
+  }
+  return <Profile imageLink={imageLink} membersData={data} />;
+};
+
+export async function getServerSideProps(context) {
+  const { params } = context;
+  const imgLink = getImgURL(params.id);
```

Comment:
NIT: if you name the variable as `imageLink`, line 27 can be just `return { props:{ imageLink, data } }`.

Rename `returnData` to `data`.